File: frontend/src/api/customer.ts
import request from './request'

export interface Customer {
  id?: number
  name: string
  company: string
  phone: string
  email: string
  address: string
  longitude?: number
  latitude?: number
  source: string
  level: string
  remark: string
  createdAt?: string
}

export const getCustomers = (params?: any) => request.get('/customers', { params })
export const getCustomer = (id: number) => request.get(`/customers/${id}`)
export const createCustomer = (data: Customer) => request.post('/customers', data)
export const updateCustomer = (id: number, data: Customer) => request.put(`/customers/${id}`, data)
export const deleteCustomer = (id: number) => request.delete(`/customers/${id}`)

export const getDashboardStats = () => request.get('/dashboard/stats')

File: backend/src/main/java/com/crm/entity/Customer.java
package com.crm.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("customers")
public class Customer {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    @TableField("name")
    private String name;
    
    @TableField("company")
    private String company;
    
    @TableField("phone")
    private String phone;
    
    @TableField("email")
    private String email;
    
    @TableField("address")
    private String address;
    
    @TableField("longitude")
    private Double longitude;
    
    @TableField("latitude")
    private Double latitude;
    
    @TableField("status")
    private String status;
    
    @TableField("source")
    private String source;
    
    @TableField("remark")
    private String remark;
    
    @TableField("created_at")
    private LocalDateTime createdAt;
    
    @TableField("updated_at")
    private LocalDateTime updatedAt;
}
